components/Sidebar/ClientSidebar.tsx

```
@@ -263,117 +334,162 @@ function ClientSidebarContent({
+                  // Replace the optimistic chat with the real one but preserve UI state
+                  const updatedChats = [...prevChats];
+                  const existingChat = prevChats[optimisticIdIndex];
+
+                  // Ensure module info is preserved if it exists in the optimistic chat
+                  // but missing in the server response
+                  const moduleInfo =
+                    newChat.module || (newChat.moduleId && existingChat.module)
+                      ? {
+                          id: newChat.moduleId || existingChat.moduleId,
+                          name:
+                            (newChat.module && newChat.module.name) ||
```

Biome (1.9.4)

[error] Change to an optional chain.

Unsafe fix: Change to an optional chain.


(lint/complexity/useOptionalChain)